### PR TITLE
Add tests for a shopping cart functionality

### PR DESCRIPTION
**Pull request by AntonZhdanov**

### CODE REVIEW

**Comment by AntonZhdanov** [comment]
TestWithCoverage

File: src/test/java/com/example/bookstore/controller/CategoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+    ) throws SQLException {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+        teardown(dataSource);
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
+                    new ClassPathResource("database/categories/add-categories.sql"));
```
Comment (vovasalyha) [inline src/test/java/com/example/bookstore/controller/CategoryControllerTest.java:62]: you can use annotation based approach too

File: src/test/java/com/example/bookstore/controller/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+            @Autowired DataSource dataSource
+    ) {
+        teardown(dataSource);
+    }
+
+    @SneakyThrows
+    private static void teardown(DataSource dataSource) {
+        try (Connection connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection,
```
Comment (vovasalyha) [inline src/test/java/com/example/bookstore/controller/CategoryControllerTest.java:89]: you can use annotation based approach too

File: src/test/java/com/example/bookstore/service/CartItemServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    @DisplayName("Save with valid parameters calls save method once")
+    void save_ValidParameters_SaveCalledOnce() {
+        itemRequestDto.setBookId(bookId);
+
+        when(bookRepository.findById(bookId)).thenReturn(Optional.of(book));
+        when(cartItemMapper.toModel(itemRequestDto)).thenReturn(cartItemMock);
+
+        cartItemServiceImpl.save(itemRequestDto, shoppingCart);
+
+        verify(cartItemRepository, times(1)).save(any(CartItem.class));
```
Comment (vovasalyha) [inline src/test/java/com/example/bookstore/service/CartItemServiceImplTest.java]: times(1) is by default, you don't need to specify it.

File: src/test/resources/database/cartitems/add-1984-to-cartitems.sql [hunk 4]
```diff
@@ -0,0 +1 @@
+insert into cart_items(id, shopping_cart_id, book_id, quantity, is_deleted) values (1, 1, 1, 2, false);
```
Comment (vovasalyha) [inline src/test/resources/database/cartitems/add-1984-to-cartitems.sql]: add empty lines where needed

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ, but see a couple of comments

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**